bee-api/bee-rest-api/src/endpoints/routes/api/v1/message_children.rs

```diff
@@ -25,24 +20,19 @@ fn path() -> impl Filter<Extract = (MessageId,), Error = warp::Rejection> + Clon
         .and(warp::path::end())
 }
 
-pub(crate) fn filter<B: StorageBackend>(
-    public_routes: Box<[String]>,
-    allowed_ips: Box<[IpAddr]>,
-    tangle: ResourceHandle<Tangle<B>>,
-) -> BoxedFilter<(impl Reply,)> {
+pub(crate) fn filter<B: StorageBackend>(args: Arc<ApiArgs<B>>) -> BoxedFilter<(impl Reply,)> {
     self::path()
         .and(warp::get())
-        .and(has_permission(ROUTE_MESSAGE_CHILDREN, public_routes, allowed_ips))
-        .and(with_tangle(tangle))
+        .and(with_args(args))
         .and_then(message_children)
         .boxed()
 }
 
 pub async fn message_children<B: StorageBackend>(
     message_id: MessageId,
-    tangle: ResourceHandle<Tangle<B>>,
+    args: Arc<ApiArgs<B>>,
 ) -> Result<impl Reply, Rejection> {
-    let mut children = Vec::from_iter(tangle.get_children(&message_id).await.unwrap_or_default());
+    let mut children = Vec::from_iter(args.tangle.get_children(&message_id).await.unwrap_or_default());
     let count = children.len();
     let max_results = 1000;
```

Comment:
I know this is not part of your PR, but should this maybe be a `const`?

Reply:
Changed! 👍